zkevm-circuits/src/tx_circuit/sign_verify.rs

```diff
@@ -400,141 +348,213 @@ impl<F: Field, const MAX_VERIF: usize> SignVerifyChip<F, MAX_VERIF> {
         })
     }
 
+    #[allow(clippy::too_many_arguments)]
+    fn assign_rlc_le(
+        &self,
+        config: &SignVerifyConfig,
+        ctx: &mut RegionCtx<F>,
+        chips: &ChipsRef<F, NUMBER_OF_LIMBS, BIT_LEN_LIMB>,
+        name: &str,
+        q_rlc: Selector,
+        challenge: Value<F>,
+        inputs_le: impl IntoIterator<Item = Term<F>>,
+    ) -> Result<AssignedCell<F, F>, Error> {
+        let zero = chips.main_gate.assign_constant(ctx, F::zero())?;
+        let columns = config.main_gate_config.advices();
+        let inputs_le = inputs_le.into_iter().collect_vec();
+        let inputs_be = iter::repeat_with(|| Term::assigned(zero.cell(), Value::known(F::zero())))
+            .take(Integer::next_multiple_of(&inputs_le.len(), &columns.len()) - inputs_le.len())
+            .chain(inputs_le.into_iter().rev())
+            .collect_vec();
+
+        let mut rlc = challenge.map(|_| F::zero());
+        for (chunk_idx, chunk) in inputs_be.chunks_exact(columns.len()).enumerate() {
+            ctx.enable(q_rlc)?;
```

Comment:
I wonder if `q_rlc` should be disabled in the last row :thinking:

Imagine you call `assign_rlc_le` consecutively.  You'd get a column `q_rlc` at 1 consecutively in all rows, but you have different RLCs, so they would mix together.  I also wonder why this is not happening; maybe `chips.main_gate.assign_constant(ctx, F::zero())?;` is taking a row, which would have `q_rlc = 0`?

Reply:
Yes, the layout with example above is actually wrong! It should be:

```diff
        // Layout (take input with length 12 as an example)
        // | q_rlc |                          rlc                        |   a   |   b   |   c   |   d    |   e    |
        // | ----- | --------------------------------------------------- | ----- | ----- | ----- | ------ | ------ |
        // |   1   |                                                   0 |     0 |     0 |     0 |  be[0] |  be[1] |
        // |   1   |                                  be[0]*r^1 +  be[1] | be[2] | be[3] | be[4] |  be[5] |  be[6] |
        // |   1   | be[0]*r^6  + be[1]*r^5  + ... +  be[5]*r^1 +  be[6] | be[7] | be[8] | be[9] | be[10] | be[11] |
-       // |   1   | be[0]*r^11 + be[1]*r^10 + ... + be[10]*r^1 + be[11] |       |       |       |        |        |
+       // |   0   | be[0]*r^11 + be[1]*r^10 + ... + be[10]*r^1 + be[11] |       |       |       |        |        |
```

And in line 392-393 is where we actually assign the last row, without enabling the `q_rlc` just like you said. Will fix the layout document soon.